pkg/sdn/plugin/subnets.go

```diff
@@ -168,43 +168,58 @@ func (master *OsdnMaster) clearInitialNodeNetworkUnavailableCondition(node *kapi
 }
 
 func (master *OsdnMaster) watchNodes() {
-	nodeAddressMap := map[types.UID]string{}
-	RunEventQueue(master.kClient.Core().RESTClient(), Nodes, func(delta cache.Delta) error {
-		node := delta.Object.(*kapi.Node)
-		name := node.ObjectMeta.Name
-		uid := node.ObjectMeta.UID
-
-		nodeIP, err := getNodeIP(node)
-		if err != nil {
-			return fmt.Errorf("failed to get node IP for %s, skipping event: %v, node: %v", name, delta.Type, node)
-		}
+	nodeInformer := master.informers.InternalKubernetesInformers().Core().InternalVersion().Nodes()
+	nodeInformer.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc: func(obj interface{}) {
```

Comment:
maybe add a helper function that takes a `shared.InformerFactory`, a "handleAddOrUpdateFunc" and a "handleDeleteFunc", and creates a `cache.ResourceEventHandlerFuncs` (doing the `DeletedFinalStateUnknown` handling) and calls `AddEventHandler` on it?